### PR TITLE
Address clippy lints

### PR DESCRIPTION
**Pull request by SanchithHegde**

### CODE REVIEW

File: src/processor.rs [hunk 1]
```diff
@@ -39,18 +39,18 @@ pub async fn handle_request(
...
             let mut response = HttpResponse::InternalServerError();
-            apply_headers(&mut response, &headers);
+            apply_headers(&mut response, headers);
             return response.finish();
         }
     };
 
     let mut response = HttpResponse::Ok();
-    apply_headers(&mut response, &headers);
+    apply_headers(&mut response, headers);
```
Comment (sansyrox) [inline src/processor.rs:53]: Hi @SanchithHegde ,

Thank you for the PR. Why did you remove the `&` here?

Comment (SanchithHegde) [reply src/processor.rs:53]: @sansyrox `clippy` throws this warning:

```
warning: this expression borrows a reference (`&std::sync::Arc<dashmap::DashMap<std::string::String, std::string::String
>>`) that is immediately dereferenced by the compiler
  --> src/processor.rs:53:34
   |
53 |     apply_headers(&mut response, &headers);
   |                                  ^^^^^^^^ help: change this to: `headers`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrow
```

Comment (sansyrox) [reply src/processor.rs:53]: @SanchithHegde , this makes sense. I will be reviewing all the PRs today. Will merge it then. Thanks again! 😄

**Comment by sansyrox** [comment]
LGTM! 🚀
